### PR TITLE
Add class Stopwatch

### PR DESCRIPTION
**Pull request by chrisdembia**
Fixes part of issue #2559 

### Brief summary of changes

- This PR adds a class Stopwatch that provides a very easy way to record and report the duration of code.

### Testing I've completed

- I employed the class in various locations and ensured the output was as expected.

### CHANGELOG.md (choose one)

- updated...

The Doxygen for this PR can be viewed at http://myosin.sourceforge.net/?C=N;O=D; click the folder whose name is this PR's number.

---
This change is [Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/2578)

### CODE REVIEW

**Comment by chrisdembia** [comment]
@aymanhab ready for review.

File: OpenSim/Common/Test/testC3DFileAdapter.cpp [hunk 1]
```diff
@@ -21,16 +21,17 @@
...
 #include <chrono>
-#include <thread>
 #include <cmath>
+#include <cstdlib>
+#include <thread>
+#include <unordered_map>
+#include <vector>
+
+#include <OpenSim/Auxiliary/auxiliaryTestFunctions.h>
+#include <OpenSim/Common/Stopwatch.h>
```
Comment (aymanhab) [inline OpenSim/Common/Test/testC3DFileAdapter.cpp]: Any reason for reordering the headers in this PR @chrisdembia ?

Comment (chrisdembia) [reply OpenSim/Common/Test/testC3DFileAdapter.cpp]: This reordering is done automatically by ClangFormat.

Comment (chrisdembia) [reply OpenSim/Common/Test/testC3DFileAdapter.cpp]: We define the rules for ordering the includes in `.clang-format`.

**Comment by chrisdembia** [comment]
Yes, the idea is to replace all those calls. I should double-check locally about how the reported times differ.

**Comment by chrisdembia** [comment]
Would you like me to do the removal of `std::clock`, etc. in this PR? 

In some cases, it doesn't make sense to use Stopwatch, because wall time is also reported.

**Comment by aymanhab** [comment]
I mainly wanted to understand the long term plan and make sure the change is mostly internal (similar reported results). If confirmed, we can do a separate PR to change the remaining instances if appropriate.

**Comment by chrisdembia** [comment]
Okay great. Timings are comparable on testC3DFileAdapter.

master:
```
	C3DFileAdapter 'walking5.c3d' loaded in 20.288ms
	Wrote 'walking5_markers.trc' in 6.859ms
	Wrote'walking5_grfs.sto' in 1.662ms
	Read'walking5_markers.trc' and its standard in 7ms
```

this PR:
```
	C3DFileAdapter 'walking5.c3d' loaded in 27 millisecond(s)
	Wrote 'walking5_markers.trc' in 6 millisecond(s)
	Wrote'walking5_grfs.sto' in 1 millisecond(s)
	Read'walking5_markers.trc' and its standard in 8 millisecond(s)
```

**Comment by chrisdembia** [comment]
Ready for review again.

**Comment by chrisdembia** [comment]
Thank you.

**Pull request merged**